projects/ngx-quill/src/lib/quill-editor.component.ts

```diff
@@ -65,28 +66,9 @@ export interface Focus {
 export type EditorChangeContent = ContentChange & {event: 'text-change'}
 export type EditorChangeSelection = SelectionChange & {event: 'selection-change'}
 
-@Component({
-  encapsulation: ViewEncapsulation.None,
-  providers: [
-    {
-      multi: true,
-      provide: NG_VALUE_ACCESSOR,
-      // eslint-disable-next-line @typescript-eslint/no-use-before-define
-      useExisting: forwardRef(() => QuillEditorComponent)
-    },
-    {
-      multi: true,
-      provide: NG_VALIDATORS,
-      // eslint-disable-next-line @typescript-eslint/no-use-before-define
-      useExisting: forwardRef(() => QuillEditorComponent)
-    }
-  ],
-  selector: 'quill-editor',
-  template: `
-  <ng-content select="[quill-editor-toolbar]"></ng-content>
-`
-})
-export class QuillEditorComponent implements AfterViewInit, ControlValueAccessor, OnChanges, OnDestroy, Validator {
+@Directive()
+// tslint:disable-next-line:directive-class-suffix
+export abstract class QuillEditorBase implements AfterViewInit, ControlValueAccessor, OnChanges, OnDestroy, Validator {
```

Comment:
The `@Component()` decorator for `QuillEditorComponent` is bit-for-bit identical to the original, we're simply swapping the lines out with the `QuillEditorBase` declaration and its `@Directive()` decorator.

Reply:
Why it has to be a @Directive?

It feels and looks a little bit strange :D.
This is the way to go in angular if you wanna provide and abstract class?

Reply:
> Why it has to be a @directive?
> This is the way to go in angular if you wanna provide and abstract class?

Correct. Since Components are an extension of Directives, this allows all decorators and metadata (like `@Input`, `@HostBinding`, etc) to be inherited by the derived `@Component` class. This also provides novel extensibility benefits, which cannot be achieved any other way -- and it has no impact on performance.

Here's a compelling example:

Base Directive:
https://github.com/angular/components/blob/master/src/material/stepper/stepper.ts#L88
Derived Component:
https://github.com/angular/components/blob/master/src/material/stepper/stepper.ts#L158
Derived Component, with same base, but diff purpose:
https://github.com/angular/components/blob/master/src/material/stepper/stepper.ts#L188

Reply:
okay, then thanks :).

I think then we can go with it.